panaroo/integrate.py

```diff
@@ -0,0 +1,272 @@
+        y['paralog'] = int(y['paralog'])
+        
+        y['longCentroidID'] = list(y['longCentroidID'])
+        y['seqIDs'] = list(y['seqIDs'])
+    
+    single_gml.graph.update({'isolateNames' : 'x'}) # isolateName assigned x as list(x) in merged graph
+    nx.write_gml(single_gml, output_dir + "final_graph.gml")
+    
+    return single_gml
+
+
+def main(): #Takes a single GFF input, generates a graph and merges with a pre-existing graph
+    args = get_options()
+
+    # make sure trailing forward slash is present
+    args.output_dir = os.path.join(args.output_dir, "")
+    
+    # Create temporary directory
+    temp_dir = os.path.join(tempfile.mkdtemp(dir=args.output_dir), "")
+     
+    directories = args.input_dir + ' ' + temp_dir
+
+    gff_file = []
+    gff_file.append(args.input_gff)
+            
+    filename = (str(args.input_gff).split('/')[-1]).split('.')[0]
+ 
+    process_prokka_input(gff_list = gff_file, 
+                         output_dir = temp_dir, 
+                         quiet = args.quiet, 
+                         n_cpu = args.n_cpu) 
+    
+    cd_hit_out = temp_dir + "/combined_protein_cdhit_out.txt"
+    
+    run_cdhit(input_file=temp_dir + "/combined_protein_CDS.fasta",
+              output_file=cd_hit_out,
+              id=args.id,
+              quiet=args.quiet,
+              n_cpu=args.n_cpu)
+    
+    single_gml, centroid_contexts_single, seqid_to_centroid_single = generate_network(cluster_file=cd_hit_out + ".clstr",
+                                                                                    data_file=temp_dir + "/gene_data.csv",
+                                                                                    prot_seq_file=temp_dir + "/combined_protein_CDS.fasta",
+                                                                                    all_dna=args.all_seq_in_graph)
+    
+    reformat_network(single_gml = single_gml,
+                     output_dir = temp_dir,
+                     isolateName = filename)
+    
+    merge_command = 'panaroo-merge -d ' + directories
+    merge_command += ' -o ' + args.output_dir
+    merge_command += ' -f ' + str(args.family_threshold)
+    merge_command += ' --len_dif_percent ' + str(args.len_dif_percent)
+    
+    if args.merge_paralogs == True:
+        merge_command += ' ' + str(args.merge_paralogs)
+    merge_command += ' --length_outlier_support_proportion ' + str(args.length_outlier_support_proportion)
+    merge_command += ' --min_edge_support_sv ' + str(args.min_edge_support_sv)
+    if args.aln is not None:
+        merge_command += ' -a ' + str(args.aln)
+    merge_command += ' --aligner ' + args.alr
+    merge_command += ' --core_threshold ' + str(args.core)
+    merge_command += ' -t ' + str(args.n_cpu)
+    if args.quiet == True:
+        merge_command += ' ' + str(args.quiet)
+    #merge_command += ' ' + str(args.version)
+    
+    subprocess.run(merge_command, shell = True)
```

Comment:
Add `check=True`

Reply:
Although, ideally this will be removed and the module imported

Reply:
removed